MAINT Parameters validation for sklearn.model_selection.cross_validate

dmitrylala opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Towards https://github.com/scikit-learn/scikit-learn/issues/24862.

#### What does this implement/fix? Explain your changes.

Automatic parameters validation for [sklearn.model_selection.cross_validate](https://github.com/scikit-learn/scikit-learn/blob/9aaed4987/sklearn/model_selection/_validation.py#L49)

Also removed None option for pre_dispatch parameter in docstring because it will cause errors according to source code of [joblib.Parallel](https://github.com/joblib/joblib/blob/a007d799a5344564a31c91648bbbfcacc646d3b1/joblib/parallel.py#L440)

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

dmitrylala committed:
MAINT Parameters validation for sklearn.model_selection.cross_validate

glemaitre commented on sklearn/tests/test_public_functions.py:
```diff
@@ -231,6 +231,7 @@ def _check_function_param_validation(
     "sklearn.metrics.top_k_accuracy_score",
     "sklearn.metrics.zero_one_loss",
     "sklearn.model_selection.train_test_split",
+    "sklearn.model_selection.cross_validate",
```
Could you make it alphabetically ordered?

dmitrylala committed:
Added sparse test and validation for cross_validate

Added sparse type for X matrix in cross_validate
and additional fixes in parameters validation.

dmitrylala commented on sklearn/model_selection/_validation.py:
```diff
@@ -46,6 +52,24 @@
+        "n_jobs": [int, None],
+        "verbose": ["verbose"],
+        "fit_params": [dict, None],
+        "pre_dispatch": [int, str],
```
Thanks for the suggestions here and above, commited.

dmitrylala commented on sklearn/model_selection/_validation.py:
```diff
@@ -46,6 +52,24 @@
+@validate_params(
+    {
+        "estimator": [HasMethods("fit")],
+        "X": ["array-like"],
```
Added test and updated docs.

dmitrylala committed:
Merge branch 'scikit-learn:main' into param-val-cross_validate

jeremiedbb committed:
cln tests now covered by common tests

jeremiedbb committed:
nitpick

jeremiedbb reviewed (approved):
I removed old tests that are now covered by the common tests.
LGTM. Thanks @dmitrylala
